packages/block-editor/src/components/block-list-appender/style.scss

```diff
@@ -16,7 +16,7 @@
 
 	// Add a uniform margin around the block.
 	// This can hopefully avoid havoc in flex containers.
-	margin: $grid-unit-10;
+	margin: 0 $grid-unit-10;
```

Comment:
I actually think that margin should be removed entirely, it's not the appender's decision to space itself to its adjacent blocks, it should be the responsibility of its container layout. (think gap) 

I actually thought I have removed this margin when I introduced the "flex" layout. Maybe I missed some places?

Reply:
One place you could see this margin fail today is:

- Insert a columns block,
- Insert an image placeholder in the first column
- Notice the appender after the placeholder, it should be left aligned by the edge of the columns block without any margin.

Conceptually speaking, we should be working on removing the responsibility to position the appender from the appender component itself and move into its layout (containers)

Reply:
> Conceptually speaking, we should be working on removing the responsibility to position the appender from the appender component itself and move into its layout (containers)

Good point. I'll take a stab. Thanks Riad!

Reply:
This works well yes. I think there's still a small issue with the Navigation block when there's only a single menu item, the gap is not applied. Ideally the block should be refactored to use the flex layout (which we'll probably do later) but in the meantime if you want to add (or remove) some specific styles to that block to handle that, that's fine.

Reply:
You'll be happy to know that it actually does use flex:

<img width="643" alt="Screenshot 2021-08-04 at 16 43 06" src="https://user-images.githubusercontent.com/1204802/128201520-df045cb7-a254-4cb2-8d6d-0ff323c75762.png">

... and I have this one handy to move from using margins to using `gap`: https://github.com/WordPress/gutenberg/pull/32367

> This works well yes. I think there's still a small issue with the Navigation block when there's only a single menu item, the gap is not applied.

Are you referring to this state:

<img width="374" alt="Screenshot 2021-08-04 at 16 44 10" src="https://user-images.githubusercontent.com/1204802/128201728-d0e02f0b-0de1-4710-a70b-3bd42e9494f0.png">

or this state?

<img width="639" alt="Screenshot 2021-08-04 at 16 44 56" src="https://user-images.githubusercontent.com/1204802/128201873-d56465ed-6eab-4fda-a157-47a6875ffa56.png">

In the second case, it's accurate, as when the navigation block doesn't have a background color, individual menu items also don't have any padding applied. That was done in #30805 to enable more patterns. You can still force a padding using global styles, and hopefully in the future an inspector control, but as far as defaults go, menu items are paddingless by default.

Reply:
I was referring to the last one and in my test I didn't see any gap between "Properties" and the appender. Might be a theme specific issue (2021)